src/cfml/system/services/ServerService.cfc

```diff
@@ -2348,24 +2353,43 @@ component accessors="true" singleton {
 		}
 	}
 
+	/**
+	 * Find out if a given Process ID (PID) is a running java service
+	 * @pidStr.hint PID to test on
+ 	 **/
+	  function isProcessAlive(required pidStr) {
+		var result = "";
+		var timeStart = millisecond(now());
+		try{
+			if (fileSystemUtil.isWindows() ) {
+				cfexecute(name='cmd', arguments='/c tasklist /FI "PID eq #pidStr#"', variable="result"  timeout="10");
+				consoleLogger.debug("Check alive Windows mode. Pid: #pidStr# in #millisecond(now()) - timeStart#ms");
+			} else if (fileSystemUtil.isMac() || fileSystemUtil.isLinux() ) {
+				cfexecute(name='ps', arguments='-p #pidStr#', variable="result" , timeout="10");
+				consoleLogger.debug("Check alive Linux/Unix mode. Pid: #pidStr# in #millisecond(now()) - timeStart#ms" );
+			} else {
+				consoleLogger.debug("Unsuported OS: Check alive for Pid: #pidStr# = return false");
+			}
+			if (findNoCase("java", result) > 0 && findNoCase(pidStr, result) > 0) return true;
+		} catch ( any e){
+			consoleLogger.debug(e);
```

Comment:
The formatting of the whole cfcatch was terrible.  Change it to just output `e.message` or something.